### DISCUSSION TITLE
Toggle Scheduler via API?

### DISCUSSION THREAD

**Discussion by TonyDeStefano** [jhuckaby/Cronicle#395] [Q&A]
Hi! When setting up an API Key, I see "Toggle Scheduler" as a privilege option. But I can't see anything in the docs or the code about how to perform this API call. We would really love to be able to toggle it off during our scale-up and scale-down events automatically. Can you point me in the right direction for that?

Thanks!
Tony

![image](https://user-images.githubusercontent.com/5439786/112393340-25253a80-8cb8-11eb-82b5-0128461d6672.png)

**Comment by mikeTWC1984** [comment]
Should be /api/app/update_master_state
Request body should contain this json:
```json
{ "enabled": true}
```
or false to disable

**Reply by TonyDeStefano** [reply]
Awesome, thank you!

**Comment by TonyDeStefano** [comment]
Can you point me to the section of code that handles this? We are having a hard time finding any reference to update_manager_state in the source. Thanks!

**Reply by mikeTWC1984** [reply]
Sorry, it's actually called  update_master_state
https://github.com/jhuckaby/Cronicle/blob/cdf3a2ee51a64396dbcaa172d84824893e742430/lib/api/admin.js#L275

**Reply by TonyDeStefano** [reply]
There it is! Thanks again!

**Comment by tushar9953** [comment]
@jhuckaby @mikeTWC1984 
looking for your help. Is there any API by which we can check whether the scheduler is enabled on disabled? Thanks in advance.

**Reply by jhuckaby** [reply]
See [get_master_state](https://github.com/jhuckaby/Cronicle/blob/master/docs/APIReference.md#get_master_state), available in v0.9.23.

**Comment by mikeTWC1984** [comment]
There is no specific api for that, but as a work around you can use login. It should return "app" object which would contain info abut state. E.g.
```bash
curl -X POST -H "Content-Type: application/json" -d '{"username":"admin", "password":"admin"}' http://localhost:3012/api/user/login | jq '.state'
```

But I agree this would be useful to have some "healthcheck" endpoint with such info.

**Comment by jhuckaby** [comment]
Please don't use the user login API for this purpose, as it creates a new session every time.

I added new APIs in [v0.9.23](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.23) to address this:

- [get_master_state](https://github.com/jhuckaby/Cronicle/blob/master/docs/APIReference.md#get_master_state)
- [update_master_state](https://github.com/jhuckaby/Cronicle/blob/master/docs/APIReference.md#update_master_state) (this API already existed, but I added docs for it).
